### PR TITLE
[10.0][IMP] shopinvader_promotion_rule: let possibility to disable promotion rule by backend

### PR DESCRIPTION
**Pull request by acsonefho**
Add a boolean on the backend to set if promotion rules should be applied or not.
The unit test only check the specific behavior.
The goal is not to test full `sale_promotion_rule` module.

### CODE REVIEW

File: shopinvader_promotion_rule/models/shopinvader_backend.py [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
+from odoo import api, fields, models
 from odoo.addons.queue_job.job import job
 
 
 class ShopinvaderBackend(models.Model):
     _inherit = "shopinvader.backend"
 
+    promotion_rule_disabled = fields.Boolean(
+        string="Disable promotion rule",
+        default=False,
```
Comment (rousseldenis) [inline shopinvader_promotion_rule/models/shopinvader_backend.py]: Remove this, default.

File: shopinvader_promotion_rule/models/shopinvader_backend.py [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
+        help="Set to True to disable promotion rules for this backend",
```
Comment (rousseldenis) [inline shopinvader_promotion_rule/models/shopinvader_backend.py]:
```suggestion
        help="Check this to disable promotion rules for this backend",
```

Set to True is too technical

**Review by rousseldenis — CHANGES_REQUESTED** [review]
LGTM. Minor remarks

**Review by lmignon — APPROVED** [review]
LGTM once comments from @rousseldenis are taken into account

**Review by rousseldenis — APPROVED** [review]
Code review

**Pull request merged**